RecoPixelVertexing/PixelLowPtUtilities/bin/ClusterShapeAnalyzer.cc

```diff
@@ -0,0 +1,322 @@
+                    std::ostringstream histName;
+                    histName << "hspc_" << is << "_" << ix << "_" << iy;
+                    TH2F * histo = (TH2F *) resFile.Get(histName.str().c_str());
+                    
+                    std::cerr << ".";
+                    
+                    // Initial guess
+                    std::vector<Center> c(2);
+                    c[0].n = 0;
+                    c[0].x[0] = ix+1;
+                    c[0].x[1] = iy+1;
+                    
+                    c[1].n = 1;
+                    c[1].x[0] =-ix-1;
+                    c[1].x[1] =-iy-1;
+                    
+                    flag << subName.at(is) << "_" << ix << "_" << iy;
+                    
+                    process(histo,c, flag.str(), print);
+                    
+                    if(histo->Integral() >= MinEntries) {
+                        // Fix barrel_0_0
+                        if(is==0 && ix==0 && iy==0) {
+                            c[0].lim[1][0] = 0.;
+                            c[1].lim[1][1] = 0.;
+                        }
+                        
+                        outFile << is << " " << ix << " " << iy;
+                        
+                        for(int b=0; b<2; b++) {
+                            for(int d=0; d<2; d++) {
+                                outFile << " " << c[b].lim[d][0] << " " << c[b].lim[d][1];
+                            }
+                        }
+                        
+                        for(int b=0; b<2; b++) {
+                            if(c[b].area > 0 )
+                                outFile << " " << c[b].n / c[b].area;
+                            else
+                                outFile << " " << 0.;
+                            
+                            outFile << " " << c[b].n;
+                        }
+                        
+                        outFile << std::endl;
+                    }
+
+                    if(mapFile)
+                        *mapFile << ix << " " << iy << " " << histo->Integral() << std::endl;
+                }
+                outFile << std::endl;
+                
+                if(mapFile)
+                    *mapFile << ix << " " << eyMax+1 << " " << 0 << "\n" << std::endl;
+                
+                std::cerr << std::endl;
+            }
+
+            if(mapFile) {
+                for(int iy=0; iy<=eyMax+1; iy++)
+                    *mapFile << exMax+1 << " " << iy << " " << 0 << std::endl;
+                *mapFile << std::endl << std::endl;
+            }
+            
+            outFile << std::endl;
+        }
+    }
+    
+private:
+    static float distance(const Center& c, const Point& p)
+    {
+        float x[2];
+        
+        for(int d=0; d<2; d++)
+            x[d] = std::fabs(p.x[d] - c.x[d]);
+        
+        return x[0]*x[0] + x[1]*x[1];
+    }
+
+    void getVectors(const std::vector<Point>& points, TH1D * x[2][2]) const
```

Comment:
please use a std::array<std::array<TH1D, 2> ,2 >& to make the second argument safer